### PR TITLE
bsd adding setproctitle (setproctitle_fast for FreeBSD too).

### LINKED ISSUE #2197: Add setproctitle
FreeBSD:

```     
#include <sys/types.h>
#include <unistd.h>

void setproctitle(const	char *fmt, ...);

void setproctitle_fast(const char *fmt,	...);
```

Certain BSD's have setproctitle to change the process title, and other unices have different ways of changing the process title (as shown in ps and such).

For FreeBSD see: https://www.freebsd.org/cgi/man.cgi?query=setproctitle&apropos=0&sektion=0&manpath=FreeBSD+13.0-RELEASE+and+Ports&arch=default&format=html

I'm not sure what's needed to get support for setproctitle, at least on platforms where it's supported. I'm not proficient at interacting with C from Rust but hopefully I can still help.

### PR DESCRIPTION
**Pull request by devnexen**
Closes #2197

### CODE REVIEW

**Comment by JohnTitor** [comment]
cc @semarie

File: src/unix/bsd/netbsdlike/openbsd/mod.rs [hunk 1]
```diff
@@ -1566,6 +1566,7 @@ extern "C" {
 
     // Added in `OpenBSD` 5.5
     pub fn explicit_bzero(s: *mut ::c_void, len: ::size_t);
+    pub fn setproctitle(fmt: *const ::c_char, ...);
```
Comment (semarie) [inline src/unix/bsd/netbsdlike/openbsd/mod.rs:1569]: mostly cosmetic remark: the position of the function addition is a bit misleading as it seems it is part of a block "Added in `OpenBSD` 5.5" (which isn't right for setproctitle).

starting a new block with a newline might be preferable

File: src/unix/bsd/netbsdlike/netbsd/mod.rs [hunk 2]
```diff
@@ -2113,6 +2113,7 @@ extern "C" {
 
     // Added in `NetBSD` 7.0
     pub fn explicit_memset(b: *mut ::c_void, c: ::c_int, len: ::size_t);
+    pub fn setproctitle(fmt: *const ::c_char, ...);
```
Comment (semarie) [inline src/unix/bsd/netbsdlike/netbsd/mod.rs:2116]: same remark than for OpenBSD. the function isn't "Added in `NetBSD` 7.0"

**Review by semarie — COMMENTED** [review]
globally fine with me, just two small nits

**Review by JohnTitor — APPROVED** [review]

**Comment by JohnTitor** [comment]
Thanks both, @bors r+

**Pull request merged**